### PR TITLE
Add mixcloud support

### LINKED ISSUE #10: Add Mixcloud support
This might need some official API..

### PR DESCRIPTION
**Pull request by faejr**
This adds mixcloud support by utilizing mixcloud-downloader.com 

This should close #10

### CODE REVIEW

File: Song.js [hunk 1]
```diff
@@ -238,6 +240,44 @@ class Song
...
+	{
+		var resolveUrl = this.url.replace("https://www.", "https://api.");
+		https.get(resolveUrl, (res) => {
+			var data = "";
+			res.setEncoding("utf8");
+			res.on("data", function(chunk) { data += chunk; });
+			res.on("end", () => {
+				if (data == "") {
+					console.log("There is no response for that URL. Resolve URL = " + urlResolve);
+					callback(this);
```
Comment (codecat) [inline Song.js]: This should be `this.callback(this);`

File: Song.js [hunk 2]
```diff
@@ -238,6 +240,44 @@ class Song
...
+					return;
+				}
+
+				var obj = JSON.parse(data);
+				if (obj.errors !== undefined && obj.errors.length > 0) {
+					console.log("Mixcloud fetch error", obj.errors);
+					return;
```
Comment (codecat) [inline Song.js]: This should also run the callback.

**Review by codecat — CHANGES_REQUESTED** [review]

File: Song.js (hunk 1 above)
Comment (faejr) [reply Song.js]: Ah I merely copy pasted the code from soundcloud. I'll fix this while I'm at it

**Review by codecat — APPROVED** [review]

**Comment by codecat** [comment]
Thanks ❤️

**Pull request merged**